FileXT/value.h

```diff
@@ -21,6 +21,15 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE  *
  * SOFTWARE.                                                                      *
  **********************************************************************************/
+ /**
+ *  Modifications:
+ *
+ * Jan 14 2022 (xoorath): Linux Port
+ *     Changed as<T> functions to named functions (like: asFloat)
+ *     The inline declarations previously used a non-compliant feature of msvc.
+ *     See: https://stackoverflow.com/questions/3052579/explicit-specialization-in-non-namespace-scope/3052604#3052604
```

Comment:
Note: if we just declare the as functions at namespace scope we will have another issue with the order of function declaration. I think we would have to move practically everything out of class declaration to underneath it or an inl file. I could be wrong, though... this just seemed easier than figuring it out. I don't see an advantage to using a template for as anyways.

Reply:
> The inline declarations previously used a non-compliant feature of msvc.
Oh god, what a disaster. Also it's not my library, it was provided by X39. Now looking at its latest version, I think they did same change there:
https://github.com/arma3/sqf-value/blob/master/sqf-value/value.hpp

Reply:
Thanks for linking the file, I actually couldn't find it. Glad to see that author was of the same mind :)

I've updated to use that version of value.hpp instead of the changes I have here.